e2e/main.py

```diff
@@ -0,0 +1,82 @@
+import os
+import sys
+import time
+import logging
+import unittest
+import argparse
+
+import docker
+import portpicker
+from docker.models.containers import Container
+
+import pytezos
+
+PROJECT_ROOT = os.path.join(os.path.dirname(__file__), "../")
+
+sys.path.append(os.path.join(PROJECT_ROOT, "client"))
+from checker import *
+
+class SandboxedTestCase(unittest.TestCase):
+    def setUp(self):
+       self.docker_client = docker.from_env()
+
+       port = portpicker.pick_unused_port()
+       self.docker_container = self.docker_client.containers.run(
+         "tqtezos/flextesa:20210316",
+         command=["edobox", "start"],
+         environment={ "block_time": 1 },
+
+         ports = { "20000/tcp": port },
+         name="checker-e2e-container-{}".format(port),
+
+         detach=True,
+         remove=True,
+       )
+
+       # wait some time for the node to start
+       time.sleep(10)
```

Comment:
I think the docker API exposes a status field which we might be able to use here in the future https://docs.docker.com/engine/api/v1.41/#operation/ContainerInspect

Reply:
That'd be nice. However, I looked at the container and I don't think it exposes that info (`HEALTCHECK`, I think it is called).

However, I just looked at `pytezos` sources, and they do something like this:

```haskell
    def _wait_for_connection(cls) -> None:
        client = cls.get_client()
        while True:
            try:
                client.shell.node.get("/version/")
                break
            except requests.exceptions.ConnectionError:
                sleep(0.1)
```

I think it is a good idea, I'll do that.